flink-tests/src/test/java/org/apache/flink/test/misc/JobIDLoggingITCase.java

```diff
@@ -149,9 +149,7 @@ void testJobIDLogging(@InjectClusterClient ClusterClient<?> clusterClient) throw
                 sourceCoordinatorLogging,
                 asList(
                         "Starting split enumerator.*",
-                        "Distributing maxAllowedWatermark.*",
-                        "Source .* registering reader for parallel task.*",
-                        "Closing SourceCoordinator for source .*"));
```

Comment:
Per my understanding, 

`Distributing maxAllowedWatermark` might not be logged if
a) announcement happens after 2 checkpoints
b) watermark is not reported before 2 checkpoints

`Closing SourceCoordinator for source` might not be captured because IIRC, job cancellation does not wait for `JobMaster` shutdown and cleanup.

`Source .* registering reader for parallel task` is still there (just a diff view artifact I guess)

Reply:
How could we restore the test coverage for the `SourceCoordinator`? 🤔 Maybe the easiest way would be to add some logging assertions to the `SourceCoordinator`'s unit tests, instead of relaying on one big `JobIDLoggingITCase` to cover everything? WDYT?

Reply:
I tried the unit test approach but it didn't work: an RPC system should be created and should receive messages instead of direct method invocations.
Which could be done I guess, but I don't it worth it.